### PR TITLE
CSP-1253 Add reason for joining section which shows on basis of condition

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/_ReasonForJoining.cshtml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿@model string?
```
Comment (DevenKShah) [inline src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/_ReasonForJoining.cshtml:1]: This value should always exist, so need not be nullable

File: src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/MemberProfile.cshtml [hunk 2]
```diff
@@ -10,7 +11,10 @@
             await Html.RenderPartialAsync("~/Views/MemberProfile/_MemberInformationSection.cshtml", Model.MemberInformation);
 
             await Html.RenderPartialAsync("~/Views/MemberProfile/_AreasOfInterestSection.cshtml", Model.AreasOfInterest);
-
+            @if (Model.MemberInformation.MaturityStatus == MemberMaturityStatus.New && Model.MemberInformation.UserRole == Role.Apprentice)
```
Comment (DevenKShah) [inline src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/MemberProfile.cshtml:14]: I would check user role first

File: src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/_ReasonForJoining.cshtml (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/_ReasonForJoining.cshtml:1]: Ok Employer user may not have this value, so it is fine.

File: src/SFA.DAS.Admin.Aan.Web/Controllers/ManageMembers/MemberProfileController.cs [hunk 3]
```diff
@@ -93,6 +95,11 @@ private async Task<AmbassadorProfileViewModel> GetViewModel(Guid id, Guid adminU
         ambassadorProfileViewModel.RemoveMember.FirstName = memberProfiles.FirstName;
         ambassadorProfileViewModel.RemoveMember.RouteLink = Url.RouteUrl(RouteNames.RemoveMember, new { id })!;
 
+        if (ambassadorProfileViewModel.MemberInformation.MaturityStatus == MemberMaturityStatus.New && ambassadorProfileViewModel.MemberInformation.UserRole == Role.Apprentice && memberProfiles.Profiles.Any(x => x.ProfileId == ProfileIds.ReasonToJoinAmbassadorNetwork))
```
Comment (DevenKShah) [inline src/SFA.DAS.Admin.Aan.Web/Controllers/ManageMembers/MemberProfileController.cs:98]: Check for role should be first condition

File: src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/MemberProfileControllerTests.cs [hunk 4]
```diff
@@ -209,6 +211,58 @@ public async Task Get_MemberWithNoActivity_ShouldReturnExpectedValueForActivitie
         });
     }
 
+    [Test, AutoData]
+    public async Task Get_NewApprenticeAccount_ShouldReturnExpectedValueForReasonForJoining(MemberProfileResponse memberProfileResponse, GetProfilesResult profilesResult, string reasonToJoin, Mock<ISessionService> _sessionServiceMock, Mock<IOuterApiClient> _outerApiClientMock, CancellationToken _cancellationToken)
```
Comment (DevenKShah) [inline src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/MemberProfileControllerTests.cs:215]: Coverage for scenario where Apprentice user who is not new should be added as well.

File: src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/MemberProfile.cshtml (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.Admin.Aan.Web/Views/MemberProfile/MemberProfile.cshtml]: Done

File: src/SFA.DAS.Admin.Aan.Web/Controllers/ManageMembers/MemberProfileController.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.Admin.Aan.Web/Controllers/ManageMembers/MemberProfileController.cs]: Done

File: src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/MemberProfileControllerTests.cs (hunk 4 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/MemberProfileControllerTests.cs:215]: Done

**Review by DevenKShah — DISMISSED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**